base/src/types.rs

```diff
@@ -40,95 +27,25 @@ impl<T: KindEnv, U: KindEnv> KindEnv for (T, U) {
     }
 }
 
-impl KindEnv for HashMap<String, TcType> {
-    fn find_kind(&self, _type_name: &SymbolRef) -> Option<RcKind> {
-        None
-    }
-}
-
 /// Trait for values which contains typed values which can be refered by name
 pub trait TypeEnv: KindEnv {
     /// Returns the type of the value bound at `id`
     fn find_type(&self, id: &SymbolRef) -> Option<&TcType>;
+
     /// Returns information about the type `id`
     fn find_type_info(&self, id: &SymbolRef) -> Option<&Alias<Symbol, TcType>>;
+
     /// Returns a record which contains all `fields`. The first element is the record type and the
     /// second is the alias type.
     fn find_record(&self, fields: &[Symbol]) -> Option<(&TcType, &TcType)>;
 }
 
-impl TypeEnv for () {
-    fn find_type(&self, _id: &SymbolRef) -> Option<&TcType> {
-        None
-    }
-    fn find_type_info(&self, _id: &SymbolRef) -> Option<&Alias<Symbol, TcType>> {
-        None
-    }
-    fn find_record(&self, _fields: &[Symbol]) -> Option<(&TcType, &TcType)> {
-        None
-    }
-}
-
-impl<'a, T: ?Sized + TypeEnv> TypeEnv for &'a T {
-    fn find_type(&self, id: &SymbolRef) -> Option<&TcType> {
-        (**self).find_type(id)
-    }
-    fn find_type_info(&self, id: &SymbolRef) -> Option<&Alias<Symbol, TcType>> {
-        (**self).find_type_info(id)
-    }
-    fn find_record(&self, fields: &[Symbol]) -> Option<(&TcType, &TcType)> {
-        (**self).find_record(fields)
-    }
-}
```

Comment:
It was necessary for allowing `(&TypeEnv, Stack)` to work but that might not be necessary anymore